starter-code/main.js

```diff
@@ -0,0 +1,62 @@
+// //******************************************************************
+// // HTML/CSS Interactions
+// //******************************************************************
+
+var memoryGame;
+
+$(document).ready(function(){
+  juanGame = new MemoryGame();
+  juanGame.shuffleCards();
+  console.log(juanGame.cards);
+
+  var html = '';
+
+  juanGame.cards.forEach(function(pic, index) {
+    var sanitizedName = pic.name.split(' ').join('_');
+
+    html += '<div class= "card" name="card_' + sanitizedName + '">';
+    html += '<div class="back"';
+    html += '    name="' + pic.name + '">';
+    html += '</div>';
+    html += '<div class="front" ';
+    html += 'style="background: url(img/' + pic.img + '") no-repeat"';
+    html += '    name="'       + pic.name +  '">';
+    html += '</div>';
+    html += '</div>';
+  });
+
+  // Add all the divs to the HTML
+
+  document.getElementById('memory_board').innerHTML = html;
+$(".back").on( "click", function(){
```

Comment:
Todo lo que contiene el click, deberia estar como un metodo en el contructor de Memory game, y pasarle por parametro el "this"